CMakeLists.txt

```diff
@@ -0,0 +1,352 @@
+  endif()
+endif()
+
+# Graphics options
+option(WITH_X11 "Build with X11 support ." ${x11_default_option_enabled})
+option(WITH_OPENGL "Build with opengl support ." ON)
+option(WITH_CAIRO "Build with cairo support ." ON)
+option(WITH_LIBPNG "Build with libpng support ." OFF)
+
+# Data storage options
+option(WITH_SQLITE "Build with SQLite support" ON)
+option(WITH_POSTGRES "Build with Postgres support" OFF)
+option(WITH_MYSQL "Build with MySQL support" OFF)
+if(WIN32)
+  option(WITH_ODBC "Build with ODBC support" OFF)
+endif()
+option(WITH_ZSTD "Build with zstd support" ON)
+option(WITH_BZLIB "Build with bzlib support" OFF)
+
+# Command-line options
+option(WITH_READLINE "Build with Readline support" OFF)
+
+# Language options
+option(WITH_FREETYPE "Build with FreeType support" ON)
+option(WITH_NLS "Build with NLS support" ${nls_default_option_enabled})
+
+# Computing options
+option(WITH_FFTW "Build with FFTW support" ON)
+option(WITH_CBLAS "Build with CBLAS support" ON)
+option(WITH_LAPACKE "Build with LAPACKE support" ON)
+option(WITH_OPENMP "Build with OpenMP support" ON)
+
+# Data format options
+option(WITH_TIFF "Build with LibTIFF support ." ON)
+option(WITH_NETCDF "Build with NetCDF support" OFF)
+option(WITH_GEOS "Build with GEOS support" OFF)
+option(WITH_PDAL "Build with PDAL support" ON)
+option(WITH_LIBLAS "Build with libLAS support" OFF)
+option(WITH_OPENDWG "Build with OpenDWG support" OFF)
+
+# Other options
+option(WITH_LARGEFILES "Build with large file support"
+       ${lfs_default_option_enabled})
+option(WITH_DOCS "Build documentation" ON)
+option(WITH_GUI "Build GUI" ON)
+option(WITH_FHS "Install with FHS" OFF)
+
+include(check_target)
+include(CheckDependentLibraries)
+
+include(get_host_arch)
+get_host_arch(BUILD_ARCH)
+
+include(get_versions)
+get_versions("include/VERSION" GRASS_VERSION_MAJOR GRASS_VERSION_MINOR
+             GRASS_VERSION_RELEASE GRASS_VERSION_DATE)
+
+set(GRASS_VERSION_NUMBER
+    ${GRASS_VERSION_MAJOR}.${GRASS_VERSION_MINOR}.${GRASS_VERSION_RELEASE})
+message(STATUS "GRASS_VERSION_NUMBER  =  '${GRASS_VERSION_NUMBER}'")
+set(GRASS_VERSION_UPDATE_PKG "0.2")
+
+# Setup GRASS install directories
+include(GRASSInstallDirs)
+
+# Setup build locations.
+if(MSVC)
+  set(CMAKE_LIBRARY_OUTPUT_DIRECTORY_${CMAKE_BUILD_TYPE} ${OUTDIR}/${GRASS_INSTALL_LIBDIR})
+elseif(NOT CMAKE_LIBRARY_OUTPUT_DIRECTORY)
+  set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${OUTDIR}/${GRASS_INSTALL_LIBDIR})
+endif()
+if(MSVC)
+  set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY_${CMAKE_BUILD_TYPE} ${OUTDIR}/${GRASS_INSTALL_LIBDIR})
+elseif(NOT CMAKE_ARCHIVE_OUTPUT_DIRECTORY)
+  set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${OUTDIR}/${GRASS_INSTALL_LIBDIR})
+endif()
+
+if(APPLE)
+  if(POLICY CMP0042)
+    cmake_policy(SET CMP0042 NEW)
+  endif()
+  set(CMAKE_MACOSX_RPATH TRUE)
+endif()
+
+set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${GRASS_INSTALL_LIBDIR}")
+set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+
+#[[ Probably not needed, but leave it here in case...
```

Comment:
What is the plan for these comments and TODOs? We can leave them here, but I think we should open an issue unless you plan to address them immediately in following PRs.

Reply:
It's a rest from Rashad (2d0e741e), can probably be removed.

Reply:
Removed.